docs/modules/ROOT/pages/documentation/payara-server/public-api/openid-connect-support.adoc

```diff
@@ -251,8 +250,96 @@ public class SecurityBean {
+* retrieve the value from an EL expression defined in the `@OpenIdAuthenticationDefinition` annotation
+
+TIP: For example, the tenant can also be resolved from a cookie, which is set the first time a user loads the application; from the domain name in the URL (if different tenants use a different domain name to access the same application); from a path prefix that follows the context root and prepends all application URLs (e.g. contextroot/tenant1/index.xhtml, contextroot/tenant2/index.xhtml).
+
+Create a file `microprofile-config.properties` in your application (for a WAR application it would be in the `WEB-INF/classes/META-INF` directory), with the following contents:
+
+.microprofile-config.properties
+[source,xml]
+----
+payara.security.openid.tenant1.providerURI=<TENANT1_OPENID_PROVIDER_URI>
+payara.security.openid.tenant2.providerURI=<TENANT2_OPENID_PROVIDER_URI>
+payara.security.openid.sessionScopedConfiguration=true
+----
+
+This will provide configuration for `tenant1` and `tenant2` tenants. For each additional tenant, add a new line for its `providerURI`.
+
+Create `OpenidConfigBean` class with the `tokenEndpointURL` method. This class will be a CDI bean that injects `HttpServletRequest` to get information about which tenant to use. It will also inject `Config` to retrieve the configuration about each tenant from the `microprofile-config.properties` file:
+
+.OpenidConfigBeanEL.java
+[source,java]
+----
+@Named
+public class OpenidConfigBeanEL {
+    
+    @Inject
+    HttpServletRequest request;
+    
+    @Inject
+    Config config;
+    
+    private static final String BASE_OPENID_KEY = "payara.security.openid";
+    
+    public String getTokenEndpointURL() {
+        String tenant = getTenant(request);  // a custom method to decide which tenant to use
+        return config
+                .getOptionalValue(BASE_OPENID_KEY + "." + tenant + ".providerURI", String.class)
+                // e.g. payara.security.openid.tenant1.providerURI for "tenant1" tenant
+                .orElseGet(() -> {
+                    // read config for the "tenant1" tenant by default
+                   return config.getValue(BASE_OPENID_KEY + ".tenant1.providerURI", String.class);
+                });
+    }
+
+    private String getTenant(HttpServletRequest request) {
+        return request.getParameter("tenant"); // resolves the tenant name from a query parameter
+    }
+    
+}
+----
+
+Finally, configure the OpenID Connector using the `OpenIdAuthenticationDefinition` annotation that references the `getTokenEndpointURL()` in an EL expression:
+
+.SecurityBean.java
+[source,java]
+----
+@OpenIdAuthenticationDefinition(
+        providerURI = "#{openidConfigBean.tokenEndpointURL}",
+        clientId = CLIENT_ID_VALUE,
+        clientSecret = CLIENT_SECRET_VALUE,
+        redirectURI = "${baseURL}/Callback"
+)
+public class SecurityBean {
+}
+----
+
+
 [[logout]]
-=== Logout functionality
+== Logout Functionality
 _Since 5.2020.3_
```

Comment:
According to the new style guide, we have decided to stop using this disclaimer, so better to remove it now.